### PR TITLE
Compresses the Availability Page Vertically.

### PR DESCRIPTION
**Pull request by meagancheese**
Screenshot 2020-08-06 at 1 04 03 PM

### CODE REVIEW

File: src/main/webapp/availability.js [hunk 1]
```diff
@@ -37,9 +37,21 @@ function loadAvailabilityTable(tableDiv, timezoneOffset) {
...
     .then(tableContents => {
       tableDiv.innerHTML = tableContents;
+      makeEmptyRowsShort();
     });
 }
 
+function makeEmptyRowsShort() {
+  let rows = document.getElementsByTagName('tr');
+  for (let row of rows) {
+    if (row.firstElementChild.innerHTML === '') {
```
Comment (apluscs) [inline src/main/webapp/availability.js]: Weird check, can we just apply the classes through JSP?

File: src/main/webapp/availability.css [hunk 2]
```diff
@@ -15,3 +15,26 @@
...
+.table th {
+  border-left: 1px solid #dee2e6;
+  border-right: 1px solid #dee2e6;
+}
+
+.short {
+  height: 10px;
+}
+
+.tall {
```
Comment (apluscs) [inline src/main/webapp/availability.css:39]: l feel these are too tall in comparison to the short ones. Any chance you can make the font smaller?

File: src/main/java/com/google/sps/data/AvailabilityTimeSlotGenerator.java [hunk 3]
```diff
@@ -180,9 +180,10 @@ private static Instant getStartOrEndOfWeek(ZonedDateTime day, boolean start) {
       if (hour > 12) {
         standardHour = hour - 12;
       }
-      String formattedTime =
-          String.format("%d:%02d %s", standardHour, minute, hour < 12 ? "AM" : "PM");
-
+      String formattedTime = "";
+      if (minute == 0) {
```
Comment (apluscs) [inline src/main/java/com/google/sps/data/AvailabilityTimeSlotGenerator.java]: Comment here on why this happens.

**Review by apluscs — CHANGES_REQUESTED** [review]

File: src/main/webapp/availability.js (hunk 1 above)
Comment (meagancheese) [reply src/main/webapp/availability.js]: Done through JSP now.

File: src/main/java/com/google/sps/data/AvailabilityTimeSlotGenerator.java (hunk 3 above)
Comment (meagancheese) [reply src/main/java/com/google/sps/data/AvailabilityTimeSlotGenerator.java]: I reverted to handling everything on the JSP page (this code no longer exists).

File: src/main/webapp/availability.css (hunk 2 above)
Comment (meagancheese) [reply src/main/webapp/availability.css:39]: Made the times smaller.

File: src/main/webapp/availabilityTable.jsp [hunk 4]
```diff
@@ -20,23 +20,37 @@
...
-      <tr>
-        <c:forEach items = "${pageScope.list}" var = "day">
-          <td onclick="toggleTile(this)" data-utc="${day.get(i).utcEncoding()}" 
-              class="${day.get(i).getClassList()}">
-            ${day.get(i).time()}
-          </td>
-        </c:forEach>
-      </tr>
+      <c:choose>
+        <c:when test="${i%4 == 0}">
```
Comment (apluscs) [inline src/main/webapp/availabilityTable.jsp]: Nit: i % 4
(the spaces

File: src/main/webapp/availabilityTable.jsp [hunk 5]
```diff
@@ -20,23 +20,37 @@
...
+          <tr class='tall'>
+            <td>${pageScope.list.get(0).get(i).time()}</td>
+            <c:forEach items = "${pageScope.list}" var = "day">
+              <td onclick="toggleTile(this)" data-utc="${day.get(i).utcEncoding()}" 
+                  class="${day.get(i).getClassList()}">
+              </td>
```
Comment (apluscs) [inline src/main/webapp/availabilityTable.jsp]: Code duplication but l guess ternary doesn't work?

**Review by apluscs — APPROVED** [review]
one nit

**Review by GrantComm — APPROVED** [review]
Merge conflict needs to be resolved but otherwise LGTM

File: src/main/webapp/availabilityTable.jsp (hunk 5 above)
Comment (meagancheese) [reply src/main/webapp/availabilityTable.jsp]: Yeah, this is how JSP handles conditionals best.

File: src/main/webapp/availabilityTable.jsp (hunk 4 above)
Comment (meagancheese) [reply src/main/webapp/availabilityTable.jsp]: Done

**Review by jalexanderqed — APPROVED** [review]

**Pull request merged**